### ISSUE TITLE
Add Maven coordinates to generated feature docs

### ISSUE THREAD

**Issue by lauracowen** [OpenLiberty/open-liberty#14633]
When using Maven (the approach we encourage), the developer must add the feature into the server.xml but also as a dependency in the pom.xml. We currently only show the server.xml config in the feature docs; we should also show the maven dependency entries that would also be needed.

eg for JAX-RS (check with Alasdair)

```
        <dependency>
            <groupId>io.openliberty.features</groupId>
            <artifactId>jaxrs-2.1</artifactId>
            <type>esa</type>
            <scope>provided</scope>
        </dependency>
```

JSON-B API as a provided maven dependency:
```
<dependency>
    <groupId>javax.json.bind</groupId>
    <artifactId>javax.json.bind-api</artifactId>
    <version>1.0</version>
    <scope>provided</scope>
</dependency>


```
(similar to info provided here: https://access.redhat.com/documentation/en-us/red_hat_openshift_application_runtimes/1/html/thorntail_runtime_guide/thorntail_fractions_reference#jax_rs)

**Comment by lauracowen** [comment]
Status: We've been working on this recently (forgot we had this issue for it). Hoping to design something early next year. /cc @hlhoots @NottyCode @dmuelle @cbridgha 

If anyone cc'd above wants to rename the issue or close it as a dupe, feel free.

**Comment by malincoln** [comment]
@cbridgha is this still valid and can you provide a general sizing to complete?   Thanks

**Comment by cbridgha** [comment]
Yes this is still valid - I would put the sizing at a Medium (4-5 weeks)

- collection of all api bundles each feature requires
- generation of Maven pom fragments
- testing that coordinates exist, and are resolvable.
- Working with docs team to format correctly

**Comment by FezzyBug** [comment]
<img width="933" height="566" alt="Image" src="https://github.com/user-attachments/assets/2b246581-f88e-4c3c-8bb3-cb03abe854aa" />

Current displayed resolution to maven coordinates (for dependencies used in pom.xml) as well as their dependent APIs - render.

@lauracowen

**Comment by NottyCode** [comment]
I don't think it is obvious what happens when you add this dependency to your maven project:

```
<dependency> 
    <groupId>io.openliberty.features</groupId>
    <artifactId>batch-1.0</artifactId>
    <version>26.0.0.9</version>
    <type>esa</type>
    <scope>provided</scope>
</dependency>
```

I think it will pull the ESA down, but I think it will also enable you to compile against the batch APIs exposed which would include the spec APIs and the non-spec APIs.

In terms of this dependency:

```
<dependency> 
    <groupId>javax.batch</groupId>
    <artifactId>javax.batch-api</artifactId>
    <version>1.0.1</version>
    ‹scope>provided</scope>
</dependency>
```

this would provide the maven coordinates for the spec API, but not any other API the feature might expose. Batch might not have an IBM API, but other features do so I'd like to understand how that would appear.

Also we show how this works with maven, but not with gradle.

**Comment by cbridgha** [comment]
We can add both Maven and Gradle sections based on the dependencies above...  I think we need to hear from @cherylking or @yeekangc , etc about using the feature esa's or pom dependecies that can point to all the api's ibm or spec level....  It should make the job easier for the developer, but our guides don't follow this convention (they point to each api separately).  In any case we are close to publishing once we get the desired formatting...

**Comment by lauracowen** [comment]
Yes, I did look at this last night but I really don't know what the best approach should be. I agree that it would be best to hear from @cherylking or @yeekangc

**Comment by yeekangc** [comment]
We generate and use the platform API features for the starter projects.  I would think that's the starting point for developers.  We can discuss further.

**Comment by yeekangc** [comment]
We might not have updated all guides to follow the suggested convention.  We can follow up too.

**Comment by cbridgha** [comment]
ok for this issue - I still think its worthwhile showing the spec specific api relevant to each feature (even if its included in the super bom's you guys are using in guides...). And explaining they could point to the feature dependency pom to bring in all the transitive ibm api's, and other features etc....     so... 

```
<dependency> 
    <groupId>io.openliberty.features</groupId>
    <artifactId>batch-1.0</artifactId>
    <version>26.0.0.9</version>
    <type>pom</type>
    <scope>provided</scope>
</dependency>
```
